### PR TITLE
Update all wc-admin endpoints to v4

### LINKED ISSUE #1264: Bump all endpoints to `v4` and hook up new orders endpoint
Blocked by #1246 

As @claudiulodro originally suggested in https://github.com/woocommerce/wc-admin/issues/950#issuecomment-450925094_ we should update all endpoints because:

- We can't add any of the new endpoints into the `v3` namespace.
- It doesn't make sense to mix and match endpoint namespaces in a feature. 
- It also wouldn't really make sense to have a `v4` namespace that doesn't contain all of the existing core `products`, `orders`, etc. endpoints.

This task will involve:
* Updating all our v3 endpoints to v4.
* Updating tests to use the v4 endpoint and making sure all tests still pass.
* Connecting the new `wc/v4/reports/orders` endpoint once #1246 lands.

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes #1264 

Bumps all wc-admin endpoints to v4 and hooks up the orders page.

### Screenshots
screen shot 2019-01-16 at 5 39 38 pm

### Detailed test instructions:

1.  Go through each wc-admin page. 
2.  Make sure pages are loading without any API 404 errors and data still loads correctly.
3.  Check that the Orders report loads the correct data with the new orders endpoint hooked up.
4.  Check that orders report contains the correct coupon data.
5.  Make sure all tests are passing.

### CODE REVIEW

**Review by Aljullu — COMMENTED** [review]
Love seeing all reports taking shape. :heart_eyes: 

I did some quick testing and everything seems to be working fine except a 404 I get when I use the _Coupon Code_ filter in the _Orders_ report:

image

Replacing `v4` with `v3` in the API call route, seems to fix it.

I also added some small comments to the PR.

File: client/analytics/components/leaderboard/test/index.js [hunk 1]
```diff
@@ -102,7 +102,7 @@ describe( 'Leaderboard', () => {
 		);
 		const leaderboard = leaderboardWrapper.root.findByType( Leaderboard );
 
-		const endpoint = '/wc/v3/reports/products';
+		const endpoint = '/wc/v4/reports/products';
```
[the remark this answers is not in the document]
Comment (joshuatf) [reply client/analytics/components/leaderboard/test/index.js]: Definitely.  Updated these and a few others in https://github.com/woocommerce/wc-admin/pull/1298/commits/f50d1d9dbb8ef80b27235c4b8a5c3de1dc66eac6

File: client/analytics/report/categories/config.js [hunk 2]
```diff
@@ -43,7 +42,7 @@ export const filters = [
 				settings: {
 					type: 'categories',
 					param: 'categories',
-					getLabels: getRequestByIdString( NAMESPACE + 'products/categories', cat => ( {
+					getLabels: getRequestByIdString( '/wc/v4/products/categories', cat => ( {
```
[the remark this answers is not in the document]
Comment (joshuatf) [reply client/analytics/report/categories/config.js]: I think this came about when I was testing with different versions.  Reverted back to `NAMESPACE` in https://github.com/woocommerce/wc-admin/pull/1298/commits/f50d1d9dbb8ef80b27235c4b8a5c3de1dc66eac6

File: client/analytics/report/orders/table.js [hunk 3]
```diff
@@ -98,43 +93,43 @@ class OrdersReportTable extends Component {
...
 			} = row;
+			const { coupons, products } = extended_info;
 
-			const products = line_items
+			const formattedProducts = products
 				.sort( ( itemA, itemB ) => itemB.quantity - itemA.quantity )
 				.map( item => ( {
 					label: item.name,
-					href: 'post.php?post=' + item.product_id + '&action=edit',
+					href: 'post.php?post=' + item.id + '&action=edit',
```
[the remark this answers is not in the document]
Comment (joshuatf) [reply client/analytics/report/orders/table.js]: Went ahead and added this!  

On a side note, when clicking through to the products, it brings up my the default date preset (month to date in my case).  I think this is expected behavior, but it feels weird going from testing a different period that has product stats to one that displays 0.  Should we append the current date range or preset from the orders report to the product URL (we could also resolve in a separate issue if you think that's better).

**Comment by joshuatf** [comment]
@Aljullu Nice catch on that coupon filter!  Added in a coupons controller override in https://github.com/woocommerce/wc-admin/pull/1298/commits/0bc67d87cd829e318a91cc27f2b1b961ecdc0c75 to update the coupon namespace to `v4` and keep all consistent.

File: includes/data-stores/class-wc-admin-reports-orders-data-store.php [hunk 4]
```diff
@@ -289,8 +291,24 @@ protected function include_extended_info( &$orders_data, $query_args ) {
...
 		}
 
+		foreach ( $coupons as $coupon ) {
+			if ( ! isset( $mapped_data[ $coupon['order_id'] ] ) ) {
+				$mapped_data[ $product['order_id'] ]['coupons'] = array();
+			}
+
+			$mapped_data[ $coupon['order_id'] ]['coupons'][] = array(
+				'id'   => $coupon['coupon_id'],
+				'code' => $coupon['coupon_code'],
```
Comment (Aljullu) [inline includes/data-stores/class-wc-admin-reports-orders-data-store.php]: I think this should be wrapper around `wc_format_coupon_code()`:

```suggestion
				'code' => wc_format_coupon_code( $coupon['coupon_code'] ),
```


I have an uppercase and a lowercase code, and they are both displayed lowercase everywhere except here:
image

It should be:
image

**Review by Aljullu — APPROVED** [review]
> @Aljullu Nice catch on that coupon filter! Added in a coupons controller override in 0bc67d8 to update the coupon namespace to v4 and keep all consistent.

It works fine now, thanks!

Everything looks good to me so I'm pre-approving this, I just added a small comment about formatting the coupon code.

**Pull request merged**